src/Assert.php

```diff
@@ -77,6 +77,27 @@ public static function isArray(mixed $value, ?string $message = null): array
         return $value;
     }
 
+    /**
+     * Assert value is a non-empty array
+     * @template       T
+     * @phpstan-assert non-empty-array $value
+     *
+     * @param T              $value
+     *
+     * @return T&non-empty-array
+     */
+    public static function isNonEmptyArray(mixed $value, ?string $message = null): array
```

Comment:
"Most" method names are without the "is". Seeing webmozart has nonEmpty and empty method, this would better fit as nonEmptyArray